ml-proto/src/spec/memory.ml

```diff
@@ -91,38 +76,56 @@ let address_of_value = function
 
 (* Load and store *)
 
-let int32_mask = Int64.shift_right_logical (Int64.of_int (-1)) 32
-let int64_of_int32_u i = Int64.logand (Int64.of_int32 i) int32_mask
+let load8 mem a ext =
+  (match ext with
+  | SX -> Int32.shift_right (Int32.shift_left (Int32.of_int !mem.{a}) 24) 24
+  | _ -> Int32.of_int !mem.{a})
+
+let load16 mem a ext =
+  Int32.logor (load8 mem a NX) (Int32.shift_left (load8 mem (a+1) ext) 8)
+
+let load32 mem a =
+  Int32.logor (load16 mem a NX) (Int32.shift_left (load16 mem (a+2) NX) 16)
+
+let load64 mem a =
+  Int64.logor (Int64.of_int32 (load32 mem a)) (Int64.shift_left (Int64.of_int32 (load32 mem (a+4))) 32)
+
+let store8 mem a bits =
+  !mem.{a} <- Int32.to_int (Int32.logand bits (Int32.of_int 255))
```

Comment:
The syntax was indeed unfamiliar to me at first, so I do appreciate clarifying it. I'd suggest a comment, something like "store the least significant byte of `bits` at memory index `a`". With the &255, I was wondering whether it was perhaps doing a bitwise or of `bits` with memory such that it actually needed the high bits to be zero or something.